‘m’ is a rigid type variable bound: build failure with 8.0.1.20161022

Build log is here: https://gist.github.com/ilovezfs/c62fa40eb3f61569cb0e03f8cd3320a8

Several errors that look like

```
Control/Monad/Logger.hs:249:10: error:
    • Couldn't match type ‘m’ with ‘IdentityT m’
      ‘m’ is a rigid type variable bound by
        the instance declaration at Control/Monad/Logger.hs:249:10
      Expected type: IdentityT
                       m (Loc -> LogSource -> LogLevel -> LogStr -> IO ())
        Actual type: IdentityT
                       (IdentityT m) (Loc -> LogSource -> LogLevel -> LogStr -> IO ())
```

Commenting out https://github.com/kazu-yamamoto/logger/blob/master/monad-logger/Control/Monad/Logger.hs#L249-L264 seems to work around the problem.

--- a/monad-logger/Control/Monad/Logger.hs
+++ b/monad-logger/Control/Monad/Logger.hs
@@ -215,8 +215,8 @@ class (MonadLogger m, MonadIO m) => MonadLoggerIO m where
     --
     -- Since 0.3.10
     askLoggerIO :: m (Loc -> LogSource -> LogLevel -> LogStr -> IO ())
-    default askLoggerIO :: (Trans.MonadTrans t, MonadLogger (t m), MonadIO (t m))
-                        => t m (Loc -> LogSource -> LogLevel -> LogStr -> IO ())
+    default askLoggerIO :: (Trans.MonadTrans t, MonadLoggerIO n, m ~ t n)
+                        => m (Loc -> LogSource -> LogLevel -> LogStr -> IO ())
     askLoggerIO = Trans.lift askLoggerIO
 
 

